hbase-coprocessor/src/test/java/org/apache/omid/transaction/TestSnapshotFilter.java

```diff
@@ -226,8 +222,115 @@ public void testGetFirstResult() throws Throwable {
     }
 
     @Test(timeOut = 60_000)
-    public void testGetSecondResult() throws Throwable {
```

Comment:
Would be good to have a test that uses a scan with FirstKeyOnlyFilter that would get incorrect results without this new implementation.

Reply:
Is there a test which would have failed without this patch (i.e. one that demonstrates the need for having the visibility filtering done as a pure HBase filter)?